packages/amplify-graphql-default-value-transformer/src/validators.ts

```diff
@@ -17,8 +17,8 @@ const validateDate = (date: string): boolean => {
   // Verify the correct number of days for
   // the month contained in the date-string
   const year = Number(date.slice(0, 4));
-  const month = Number(date.slice(5, 2));
-  const day = Number(date.slice(8, 2));
+  const month = Number(date.slice(5, 7));
```

Comment:
this looks like a bug that we're fixing here?

Reply:
Yeah, I think somebody interpreted the slice api as (start, length), but it's (start, end). The effect here is that day/month were ALWAYS deserialized as 0, and we never did day value validation.

Reply:
It's tested in the AWSDate tests (e.g. ensuring that feb fails on 29 in some cases, on 30 in nov, etc.

Reply:
Previously those tests would always pass.